ext/calendar/easter.c

```diff
@@ -48,8 +49,8 @@ static void _cal_easter(INTERNAL_FUNCTION_PARAMETERS, bool gm)
 		}
 	}
 
-	if (year < 0 || year > (ZEND_LONG_MAX - 1)) {
-		zend_argument_value_error(1, "must be between 0 and " ZEND_LONG_FMT, (ZEND_LONG_MAX - 1));
+	if (year <= 0 || (double)year > max_year) {
+		zend_argument_value_error(1, "must be between 1 and " ZEND_LONG_FMT, (zend_long)max_year);
```

Comment:
`(zend_long)max_year` can also be UB because it's not necessarily representable?

Reply:
Nvm this is not true.